src/Compatibility/ApiCompat/Microsoft.DotNet.ApiCompat.Task/Microsoft.DotNet.ApiCompat.Task.csproj

```diff
@@ -2,6 +2,8 @@
 
   <PropertyGroup>
     <TargetFrameworks>$(NetToolMinimum);$(NetFrameworkToolCurrent)</TargetFrameworks>
+    <!-- Disable transitive pinning, we can't upgrade dependencies that are excluded from shipping  -->
+    <CentralPackageTransitivePinningEnabled>false</CentralPackageTransitivePinningEnabled>
```

Comment:
I did this to all the places that were contributing to APICompat task.  It should not impact the tool.

We might also want something similar for GenAPI - but I doubt we ever run that in desktop MSBuild.

Reply:
What about Microsoft.DotNet.PacakgeValidation.csproj?

Reply:
As an alternative, could we pin the problematic packages to the lower version? We are already doing this for CodeAnalysis: https://github.com/dotnet/sdk/blob/ec4dcaf0a5a316a605d90dc59fa52c0fb9949550/src/Compatibility/Directory.Packages.props#L5-L9

Reply:
We'd need to pin the entire closure of CodeAnalysis, including anything it might take a dependency on in the future.  IMO that's not a great solution - pretty fragile and difficult to maintain.

We should pin our "entrypoint" packages where we have a clear understanding of target version.  We let them define which other packages they can provide and don't do so ourselves.